lib/archethic/contracts/contract.ex

```diff
@@ -52,6 +52,26 @@ defmodule Archethic.Contracts.Contract do
     }
   end
 
+  @doc """
+  Create a contract from a transaction
+  """
+  @spec from_transaction(Transaction.t()) :: {:ok, t()} | {:error, String.t()}
+  def from_transaction(tx = %Transaction{data: %TransactionData{code: code}})
+      when is_binary(code) and code != "" do
```

Comment:
this guard will throw, should this be an `{:error, _}` instead?

Reply:
Sure, as Interpreter.parse already handle the case.